executor/write_test.go

```diff
@@ -2031,3 +2031,28 @@ func (s *testSuite) TestReplaceLog(c *C) {
 
 	tk.MustQuery(`admin cleanup index testLog b;`).Check(testkit.Rows("1"))
 }
+
+// For issue 7422.
+// There is no need to do the rebase when updating a record if the auto-increment ID not changed.
+// This could make the auto ID increasing speed slower.
+func (s *testSuite) TestRebaseIfNeeded(c *C) {
```

Comment:
Maybe need add insert on duplicate key update testcase, and check `last_insert_id()` value.

Reply:
Already have the cases at line 595 to 627.

Reply:
I means rebase auto_increment in `on duplicate key update`, current case seems not update id?